scripts/test.sh

```diff
@@ -44,16 +48,33 @@ stop_server() {
   fi
 }
 
+startedSimulator=false
+log_temp=
+test_temp_dir=
 cleanup() {
   # Kill started object server
-  stop_server
-
+  stop_server || true
+  
+  # Give the simulator a chance to go down gracefully
+  if $startedSimulator; then
+    osascript -e 'tell app "Simulator" to quit without saving' || true
+    sleep 0.25 # otherwise the pkill following will get it
+  fi
+  
   # Kill all other child processes.
   pkill -P $$ || true
 
   # Kill react native packager
   pkill node || true
   rm -f "$PACKAGER_OUT" "$LOGCAT_OUT"
+  
+  # Cleanup temp files
+  if [ -n "$log_temp" ] && [ -e "$log_temp" ]; then
+    rm "$log_temp" || true
+  fi
+  if [ -n "$test_temp_dir" ] && [ -e "$test_temp_dir" ]; then
+    rm -rf "$test_temp_dir" || true
```

Comment:
the `-f` flag IIRC means that the `|| true` or the `-e` check above shouldn't be necessary